src/strategies/jade-smrt/index.ts

```diff
@@ -0,0 +1,182 @@
+import { subgraphRequest } from '../../utils';
+import { getProvider } from '../../utils';
+import strategies from '..';
+import fetch from 'cross-fetch';
+
+
+export const author = 'drgorillamd';
+export const version = '1.0.0';
+
+const SMRT = '0xCC2f1d827b18321254223dF4e84dE399D9Ff116c';
+const SMRTR = '0x6D923f688C7FF287dc3A5943CAeefc994F97b290';
+const JADE = '0x7ad7242A99F21aa543F9650A56D141C57e4F6081';
+
+const defaultGraphs = {
+  '56': 'https://api.thegraph.com/subgraphs/name/apyvision/block-info',
+};
+
+async function getChainBlockNumber(
+  timestamp: number,
+  graphURL: string
+): Promise<number> {
+  const query = {
+    blocks: {
+      __args: {
+        first: 1,
+        orderBy: 'number',
+        orderDirection: 'desc',
+        where: {
+          timestamp_lte: timestamp
+        }
+      },
+      number: true,
+      timestamp: true
+    }
+  };
+  const data = await subgraphRequest(graphURL, query);
+  return Number(data.blocks[0].number);
+}
+
+async function getChainBlocks(
+  snapshot,
+  provider,
+  options,
+  network
+): Promise<any> {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const block = await provider.getBlock(blockTag);
+  const chainBlocks = {};
+  for (const strategy of options.strategies) {
+    if (chainBlocks[strategy.network]) {
+      continue;
+    }
+    if (blockTag === 'latest' || strategy.network === network) {
+      chainBlocks[strategy.network] = blockTag;
+    } else {
+      const graph =
+        options.graphs?.[strategy.network] || defaultGraphs[strategy.network];
+      chainBlocks[strategy.network] = await getChainBlockNumber(
+        block.timestamp,
+        graph
+      );
+    }
+  }
+
+  return chainBlocks;
+}
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+
+  const chainBlocks = await getChainBlocks(
+    snapshot,
+    provider,
+    options,
+    network
+  );
+
+  let block = await provider.getBlock(typeof snapshot === 'number' ? snapshot : 'latest');
+  let timestamp = block.timestamp;
+  let jadePrice: number = await geckoPrice(JADE, timestamp, 'binance-smart-chain');
+  let results: any = [];
+
+  for (const strategy of options.strategies) {
```

Comment:
we should limit these strategies passed into these. else this could increase the load on the server (may be one or two strategies should be good, @bonustrack any suggestions on the number? )

Reply:
I took it from the multichain (the example has 5 strat iirc). I just send you a message on discord with my former function prototypes, it was rather rewriting erc20-balanceOf/uni-LP (and resolve everything at once) than building on top of multichain/erc20-balanceOf/etc and iterating over strats.
Let me know if you want me to roll back to that implementation, happy to do so!